pkg/controller/statefulset/stateful_set_control_test.go

```diff
@@ -2839,6 +2844,10 @@ func newRevisionOrDie(set *apps.StatefulSet, revision int64) *apps.ControllerRev
 }
 
 func isOrHasInternalError(err error) bool {
+	if err == nil {
+		return false
+	}
+
 	agg, ok := err.(utilerrors.Aggregate)
-	return !ok && !apierrors.IsInternalError(err) || ok && len(agg.Errors()) > 0 && !apierrors.IsInternalError(agg.Errors()[0])
+	return !ok && apierrors.IsInternalError(err) || ok && len(agg.Errors()) > 0 && apierrors.IsInternalError(agg.Errors()[0])
```

Comment:
I'm thinking if this change is really required since we'll still need to check if any of api errors are not being returned for the method `isOrHasInternalError`.

Reply:
@pravarag 

> I'm thinking if this change is really required since we'll still need to check if any of api errors are not being returned for the method `isOrHasInternalError`.

This is quite a confusing fix...

According to the contexts and test log messages of its callers, I think the function was intended to be used to catch cases that don't throw internal errors.

However,
1. the function name does not match its functionality.
2. All callers misuse this function
https://github.com/kubernetes/kubernetes/blob/6bdda2da160043c5fcdfd25bbea9e1c1b804c9e5/pkg/controller/statefulset/stateful_set_control_test.go#L415-L417

^ The above test should fail if `scaleUpStatefulSetControl` does not return an internal error, but it succeeds if it returns nil error.

This PR fixes its implementation to match its name and fixes its use cases.